### DISCUSSION TITLE
Is there a Gallery application that showcases the look and feel of all default UI Components?

### DISCUSSION THREAD

**Discussion by futuremotiondev** [AvaloniaUI/Avalonia#10885] [General]
Looking for something similar to:

https://apps.microsoft.com/store/detail/winui-3-gallery/9P3JFPWWDZRC?hl=en-us&gl=us (MS Fluent Design / WinUI 3)
or
https://apps.microsoft.com/store/detail/syncfusion-wpf-controls-examples/9N99KDHRFF6G?hl=en-us&gl=us (Syncfusion WPF)
or
https://user-images.githubusercontent.com/13592821/166259110-0fb98120-fe34-4e6d-ab92-9f72ad7113c3.png (WPF UI)

Is there anything like this that exists for Avalonia?

**Comment by jp2masa** [comment]
https://apps.microsoft.com/store/detail/avalonia-control-gallery/9PGHN6QNJ3RP

**Comment by futuremotiondev** [comment]
Weird ... after installing that app and running it, I'm only seeing the `Overview`, `Date and Time`, and `Layout` categories on the left.

![XamlControlsGallery_xj7f1Sc0Km](https://user-images.githubusercontent.com/33441569/229374220-11896a8e-d6d9-4d31-bd15-3f1e7aa437af.png)

Is this correct or is the install glitched?

**Reply by jp2masa** [reply]
It's correct.

**Comment by futuremotiondev** [comment]
Thank you jp2masa for the link. Closing this.

**Comment by rabbitism** [comment]
This is very outdated thou.

**Comment by timunie** [comment]
@fmotion1 imo the best control library you can get at the moment is to clone the source of Avalonia and explore the samples, especially the ControlCatalog. See: https://docs.avaloniaui.net/guides/developer-guides/build-avalonia-from-source

**Reply by futuremotiondev** [reply]
Great. I will do that right away. Thanks for the help! I'm trying to figure out the advantages of Avalonia UI vs MAUI or WinUI 3. MAUI is also cross-platform and looks a lot more fleshed out. Likewise if I wanted to build a windows app only - WinUI 3 with the App SDK looks a lot more mature. Could someone explain the actual advantages of Avalonia over these other options?

**Reply by timunie** [reply]
Asking in the Avalonia forum may lead to biased answers, but however let me tell you the reasons I have chosen Avalonia over other: 

- Avalonia has overall a good performance 
- It is cross platform, even though I only use it on windows
- It renders (almost) pixel perfect on any OS. So no matter the OS (Win 10, Win 11, OSX) the UI looks always the same which is what I prefer
- The ability to easily restyle the controls to my needs

Let me also tell you why I didn't chose one of the other options: 

WPF:
- Looks like there is not much activity on MS side atm and I'm afraid support will be stopped some day
- Some bugs that I had which were not solved for years. In Avalonia I can file a PR and have a good chance to get it merged quickly
- Good overall community support IMO

MAUI:
- MAUI uses platform controls, so your App will look different acc. to the OS. I don't like this, but its a personal preference.
- Please also check out: https://www.avaloniaui.net/Blog/avalonia-ui-and-maui-something-for-everyone,ce77d8d4-8dad-45ce-ae5a-566f3786571a

WinUI3: 
- WinUI2 was already a pain in the a**
- No data validation support when I last checked and this is important for me to have
- No Win7 support which is important as some of my users still have Win7 machines (also in Avalonia not officially supported, but it just works ^^)
- When I listen to some in my bubble, most of them say that WinUI3 is not yet stable. But as I said, this is a biased option
- No option to make the App x-plat later if needed. In Avalonia this is a simple task


Last note: If you want to polish your Avalonia-App, check out https://github.com/AvaloniaCommunity/awesome-avalonia . There are some really nice styles available (I use FluentAvalonia) or you can easily roll your own styles which will look exactly as you want. So you can give your App a personal touch. 

I hope we can now welcome you in the Avalonia world ;-)
